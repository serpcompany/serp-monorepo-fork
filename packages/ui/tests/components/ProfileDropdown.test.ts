import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import { ref } from 'vue';
import ProfileDropdown from '../../components/ProfileDropdown.client.vue';
import ComponentRender from '../componentRender';

// Declare a mutable session_ variable.
let session_: { loggedIn: unknown; user: unknown; clear: unknown };

// Mock the useUserSession composable
mockNuxtImport('useUserSession', () => () => session_);

describe('ProfileDropdown Snapshot', () => {
  // Reset session_ before each test.
  beforeEach(() => {
    session_ = { loggedIn: ref(false), user: ref(null), clear: vi.fn() };
  });

  const scenarios: [
    string,
    { session: { loggedIn: unknown; user: unknown; clear: unknown } }
  ][] = [
    [
      'when logged in',
      {
        session: {
          loggedIn: ref(true),
          user: ref({
            name: 'Test User',
            email: 'test@test.com',
            image: 'https://example.com/test.jpg'
          }),
          clear: vi.fn()
        }
      }
    ],
    [
      'when not logged in',
      {
        session: {
          loggedIn: ref(false),
          user: ref(null),
          clear: vi.fn()
        }
      }
    ]
  ];

  it.each(scenarios)('renders %s correctly', async (desc, { session }) => {
    // Update the session for this scenario.
    session_ = session;

    // Render the component and generate the snapshot.
    const html = await ComponentRender(
      `ProfileDropdown ${desc}`,
      {},
      ProfileDropdown
    );
    expect(html).toMatchSnapshot();
  });
});
