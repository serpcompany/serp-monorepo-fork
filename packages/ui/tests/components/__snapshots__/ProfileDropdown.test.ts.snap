// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileDropdown Snapshot > renders when logged in correctly 1`] = `
"<div class="relative">
  <!-- Avatar trigger -->
  <div><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden rounded-full align-middle bg-elevated size-10 text-xl cursor-pointer transition-opacity hover:opacity-80"><img width="40" height="40" data-nuxt-img="" srcset="https://example.com/test.jpg 1x, https://example.com/test.jpg 2x" role="button" class="h-full w-full rounded-[inherit] object-cover" src="https://example.com/test.jpg"></span></div><!-- Slideover menu -->
  <!--v-if-->
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`ProfileDropdown Snapshot > renders when not logged in correctly 1`] = `
"<!-- Login button for logged-out users -->
<a href="/login" class="inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors py-1.5 gap-1.5 text-primary hover:bg-primary/10 focus:outline-none focus-visible:bg-primary/10 disabled:bg-transparent aria-disabled:bg-transparent dark:disabled:bg-transparent dark:aria-disabled:bg-transparent dark:text-primary-100 text-md rounded-md border border-neutral-300 px-6 font-medium">
  <!--v-if--> Login
  <!--v-if-->
</a>"
`;
